### PR TITLE
Converted PromptComponent class-based component to functional component

### LINKED ISSUE #8779: [UI] Convert `PromptComponent` class-based component to functional component
### Current Behavior
PromptComponent is a class-based component.

### Desired Behavior
PromptComponents should be a functional component.

### Implementation
Convert all the Class-based components to new updated functional components. It is important here to convert a class-based component to a functional component you may need to convert some functions as well. Check if the function is used in other components or not.

Also, make sure to use the hooks while converting them. For example, you need to covert withNotify with useNotification, useRouter hook. Please check if a hook for functional components is available and use that.

Here is a quick blog that will help you get started with : [Link](https://www.digitalocean.com/community/tutorials/five-ways-to-convert-react-class-components-to-functional-components-with-react-hooks)

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Ishwar-Zatke**
**Description:**
Converted the class-based component  of PromptCOmponent to functional based component.
 

This PR fixes 
#8779 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Ishwar-Zatke** [comment]
@Rajdip019 , Can you review the changes?

File: ui/components/PromptComponent.js [hunk 1]
```diff
@@ -60,87 +60,77 @@ const styles = (theme) => ({
...
-      subtitle : "",
-      options : []
+function PromptComponent(props) {
+  const [show, setShow] = useState(false);
+  const [title] = useState("");
+  const [subtitle] = useState("");
+  const [options] = useState([]);
+  const [promiseInfo] = useState({ resolve : null, reject : null });
+
+  useEffect(() => {
```
Comment (Rajdip019) [inline ui/components/PromptComponent.js:70]: Is this `useEffect` really necessary?

Comment (Ishwar-Zatke) [reply ui/components/PromptComponent.js:70]: It's purpose was to listen for changes in the show state, It isn't that necessary

Comment (Ishwar-Zatke) [reply ui/components/PromptComponent.js]: Have a look at the change @Rajdip019 , and is anything necessary to update there

Comment (Rajdip019) [reply ui/components/PromptComponent.js]: Looks good I guess.

Comment (Ishwar-Zatke) [reply ui/components/PromptComponent.js]: @Rajdip019  could you please perform a final review to ensure everything is in order and, if appropriate, merge the PR?

**Comment by saurabh100ni** [comment]
hey @Ishwar-Zatke 

Let's discuss this as your agenda item today in the Meshery Development Meeting,
Please add this as your agenda if you would: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

**Comment by leecalcote** [comment]
Merge conflict...

**Comment by leecalcote** [comment]
@Ishwar-Zatke a couple of merge conflicts have cropped up. Will you have a look, please?

**Pull request closed without merging**